src/main/java/appeng/util/InventoryAdaptor.java

```diff
@@ -55,69 +61,121 @@
+    public static InventoryAdaptor getAdaptor(Object te, final ForgeDirection d,
+            @MagicConstant(flagsFromClass = InventoryAdaptor.class) int flags) {
         if (te == null) {
             return null;
         }
 
-        final IBetterStorage bs = (IBetterStorage) (IntegrationRegistry.INSTANCE.isEnabled(
-                IntegrationType.BetterStorage) ? IntegrationRegistry.INSTANCE.getInstance(IntegrationType.BetterStorage)
-                        : null);
-        final IThaumicTinkerer tt = (IThaumicTinkerer) (IntegrationRegistry.INSTANCE
-                .isEnabled(IntegrationType.ThaumicTinkerer)
-                        ? IntegrationRegistry.INSTANCE.getInstance(IntegrationType.ThaumicTinkerer)
-                        : null);
+        boolean invs = (flags & ALLOW_ITEMS) != 0;
+        boolean tanks = (flags & ALLOW_FLUIDS) != 0;
+
+        final IBetterStorage bs = IntegrationRegistry.INSTANCE.getInstanceIfEnabled(IntegrationType.BetterStorage);
+        final IThaumicTinkerer tt = IntegrationRegistry.INSTANCE.getInstanceIfEnabled(IntegrationType.ThaumicTinkerer);
 
         if (tt != null && tt.isTransvectorInterface(te)) {
             te = tt.getTile(te);
         }
 
-        if (isEIOLoaded && te instanceof TileConduitBundle tcb) {
+        // spotless:off
+        if (invs && isEIOLoaded && te instanceof TileConduitBundle tcb) {
             return new AdaptorConduitBandle(tcb, d);
-        } else if (te instanceof EntityPlayer) {
+        }
+
+        if (invs && te instanceof EntityPlayer) {
             return new AdaptorIInventory(new AdaptorPlayerInventory(((EntityPlayer) te).inventory, false));
-        } else if (te instanceof ArrayList) {
+        }
+
+        if (invs && te instanceof ArrayList) {
             @SuppressWarnings("unchecked")
             final ArrayList<ItemStack> list = (ArrayList<ItemStack>) te;
 
             return new AdaptorList(list);
-        } else if (bs != null && bs.isStorageCrate(te)) {
+        }
+
+        if (invs && bs != null && bs.isStorageCrate(te)) {
             return bs.getAdaptor(te, d);
-        } else if (te instanceof TileEntityChest) {
+        }
+
+        if (invs && te instanceof TileEntityChest) {
             return new AdaptorIInventory(Platform.GetChestInv(te));
-        } else if (te instanceof ISidedInventory si) {
-            if (te instanceof TileInterface) {
-                return new AdaptorDualityInterface(new WrapperMCISidedInventory(si, d), (IInterfaceHost) te);
-            } else if (te instanceof TileCableBus) {
-                IPart part = ((TileCableBus) te).getPart(d);
-                if (part instanceof IInterfaceHost host) {
-                    return new AdaptorDualityInterface(new WrapperMCISidedInventory(si, d), host);
-                } else if (part instanceof PartP2PItems p2p) {
+        }
+
+        if (te instanceof ISidedInventory sided) {
+            if (invs && te instanceof TileInterface) {
```

Comment:
Not a fan of those `invs &&` everywhere, could we split this into two methods, one for invs and one for tanks that might return null?